src/backend/distributed/commands/utility_hook.c

```diff
@@ -1603,20 +1628,22 @@ DropSchemaOrDBInProgress(void)
 static void
 RunPreprocessMainDBCommand(Node *parsetree, const char *queryString)
 {
-	if (IsA(parsetree, CreateRoleStmt))
+	if (!IsStatementSupportedIn2PC(parsetree))
 	{
-		StringInfo mainDBQuery = makeStringInfo();
-		appendStringInfo(mainDBQuery,
-						 START_MANAGEMENT_TRANSACTION,
-						 GetCurrentFullTransactionId().value);
-		RunCitusMainDBQuery(mainDBQuery->data);
-		mainDBQuery = makeStringInfo();
-		appendStringInfo(mainDBQuery,
-						 EXECUTE_COMMAND_ON_REMOTE_NODES_AS_USER,
-						 quote_literal_cstr(queryString),
-						 quote_literal_cstr(CurrentUserName()));
-		RunCitusMainDBQuery(mainDBQuery->data);
+		return;
 	}
+
+	StringInfo mainDBQuery = makeStringInfo();
+	appendStringInfo(mainDBQuery,
+					 START_MANAGEMENT_TRANSACTION,
+					 GetCurrentFullTransactionId().value);
+	RunCitusMainDBQuery(mainDBQuery->data);
+	mainDBQuery = makeStringInfo();
+	appendStringInfo(mainDBQuery,
+					 EXECUTE_COMMAND_ON_REMOTE_NODES_AS_USER,
+					 quote_literal_cstr(queryString),
```

Comment:
Can we use deparsed query string here as we discussed earlier? See https://github.com/citusdata/citus/issues/7448.

Let's just do that for GRANT ROLE in this PR, I suppose Ozan will fix the code-path for CREATE ROLE commands in another PR.

Reply:
ping, let's use deparsed query string here

Reply:
ping, let's use deparsed query string here for all cases

Reply:
Uses DeparseTreeNode